### PR TITLE
Properly implements the "default-template" mechanism for extensions.

### PR DESCRIPTION
**Pull request by andyHa**
If a template in "default/extensions/name" and one in "extensions/name"
exists, we now only load the second (in accordance with all other places
where default templates are overwritten by ordinary ones with the same
name).

This way a default extension in a library can be provided which can be
overwritten in a product).

### CODE REVIEW

**Review by jakobvogel — APPROVED** [review]

File: src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java [hunk 1]
```diff
@@ -124,15 +125,29 @@
...
-                                           + "/.*.pasta"))
-                     .map(m -> m.group(0))
-                     .map(this::resolveToTemplateExtension)
-                     .filter(Objects::nonNull)
-                     .sorted()
-                     .collect(Collectors.toList());
+        List<TemplateExtension> extensionsForTarget = Sirius.getClasspath()
+                                                            .find(Pattern.compile("(customizations/[^/]+/)?extensions/"
+                                                                                  + Pattern.quote(target)
+                                                                                  + "/.*.pasta"))
```
Comment (sabieber) [inline src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java:131]: Extract pattern into constant so it is not always recompiled?

File: src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java [hunk 2]
```diff
@@ -124,15 +125,29 @@
...
+                                                            .map(m -> m.group(0))
+                                                            .map(this::resolveToTemplateExtension)
+                                                            .filter(Objects::nonNull)
+                                                            .toList();
+
+        // Also check for extensions provided as default - which are only loaded, if no overwrites (same path without
+        // /default/ in front) exists...
+        return Streams.concat(extensionsForTarget.stream(),
+                              Sirius.getClasspath()
+                                    .find(Pattern.compile("default/extensions/" + Pattern.quote(target) + "/.*.pasta"))
```
Comment (sabieber) [inline src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java:141]: s.a.

**Review by sabieber — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java:131]: this isn't constant as "target" is passed in - however the result of the method call is cached...

File: src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/pasta/tagliatelle/Tagliatelle.java:141]: this isn't constant as "target" is passed in - however the result of the method call is cached...

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

**Pull request merged**